protected.go

```diff
@@ -150,16 +145,16 @@ func (p *Protector) DialContext(ctx context.Context, network, addr string) (net.
 	chDone := make(chan bool)
 	go func() {
 		conn, err = p.dialContext(op, ctx, network, addr)
-		chDone <- true
-	}()
-	select {
-	case <-ctx.Done():
-		go func() {
-			<-chDone
+		select {
```

Comment:
This change avoids creating another goroutine, and saves one or two lines of code.